layer2/oscal-parser.go

```diff
@@ -0,0 +1,93 @@
+package layer2
```

Comment:
The file naming here `oscal-parser.go` makes me assume that this contains code that consumes OSCAL. However, the code here is more about producing OSCAL. Perhaps rename to `oscal_generator.go`

Also, note I specified an underscore, not a hyphen as all other `.go` files in these directories use underscores in their names, rather than hyphens.